### ISSUE TITLE
[BUG] Azure Media Services HLS URL fails to play

### ISSUE THREAD

**Issue by JohnGalt1717** [jhomlala/betterplayer#247]
**Describe the bug**
I have the following HLS URL which is a clear (No DRM or AES headers) that plays fine in a web browser: (test here: "https://ampdemo.azureedge.net/?url=https%3A%2F%2Fstbetams-usea.streaming.media.azure.net%2Fbad5c3ac-1563-408b-9850-e7190df8934b%2Finput-content-1-A.ism%2Fmanifest(format%3Dm3u8-aapl)") 

https://stbetams-usea.streaming.media.azure.net/bad5c3ac-1563-408b-9850-e7190df8934b/input-content-1-A.ism/manifest(format=m3u8-aapl)

This also works on a test app I did on Xamarin.

Here's the DASH Url that also doesn't work: https://stbetams-usea.streaming.media.azure.net/bad5c3ac-1563-408b-9850-e7190df8934b/input-content-1-A.ism/manifest(format=mpd-time-csf)

And the SmoothStreaming URL  that doesn't work: https://stbetams-usea.streaming.media.azure.net/bad5c3ac-1563-408b-9850-e7190df8934b/input-content-1-A.ism/manifest

**To Reproduce**
1. Create a new better player and pass any of the above URLs for BetterPlayer.network(url: ....)
2. Attempt to play the video.
3. Errors with a exoplayer network error. (this is in an android emluator on windows)

**Expected behavior**
Should playback the m3u8 file that is returned from the URL.

**Flutter doctor**
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 1.25.0-8.3.pre, on Microsoft Windows [Version 10.0.19042.746], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.52.1)      
[√] Connected device (3 available)

• No issues found!

**Better Player version**
- Version: 0.48

**Smartphone (please complete the following information):**
 - Device: Pixel C emulator
 - OS: Android 11 SDK 30

**Additional context**

**Comment by jhomlala** [comment]
@JohnGalt1717 These links don't work.  Seems like that you're struggling with this issue for a long time (https://github.com/google/ExoPlayer/issues/6844) . Do you have any example from ExoPlayer which works? How about iOS? Did you check it?

**Comment by JohnGalt1717** [comment]
ios just works.

What I've found is that V3 of HLS seems to work on Android for some reason even though V4 is supported. So I've hacked the code.

The other issue with Azure Media Services and the flutter video player is that the video player uses the file extension to understand what it is processing instead of looking at what it got back or the response type from the network request and it doesn't appear to be any way of telling it or getting it to use the response content type.

So I've hacked it to put on the extension + use HLS V3 and it works. Obviously not ideal.
